lib/ret_web/channels/hub_channel.ex

```diff
@@ -876,6 +876,18 @@ defmodule RetWeb.HubChannel do
     {:error, %{message: "Hub no longer accessible", reason: "closed"}}
   end
 
+  defp join_with_hub(
+         %Hub{},
+         %Account{},
+         _socket,
+         _context,
+         %{
+           hub_requires_oauth: false,
+           account_can_join: false
```

Comment:
Are there any cases where `account_can_join` is `false` but the join is _not_ denied? Wondering how this and the `join_with_hub` case directly below relate to the `can?(_account, :join_hub ...` checks in hub.ex

Reply:
`account_can_join` is derived from the `:join_hub` permission check, so it should be impossible for account_can_join to be false if the join permission is denied. The case below only covers Discord-bound hubs.

Reply:
What I mean to ask is whether a match on `account_can_join: false` is a sufficient condition for us to want to `deny_join`, as in: 
```
  defp join_with_hub(
         %Hub{},
         %Account{},
         _socket,
         _context,
         %{
           account_can_join: false
         }
       ),
       do: deny_join()
```
I'm trying to understand whether it's possible for `account_can_join` to be `false` but still fall through to the final `join_with_hub`.

Reply:
Ah, I see. I don't think we can do that because we want to reply with `require_oauth` in another scenario below.